pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go

```diff
@@ -186,6 +186,10 @@ func makePlan(
 			execinfrapb.Ordering{},
 		)
 
+		// NB: We schedule the frontier processor on the
+		// gateway node. The frontier process assumes this is
+		// the case so that it can load the current job with
+		// claim information.
```

Comment:
Could you clarify for my lack of job knowledge:
- we now assume when the processor starts, the frontier proc is on the "gateway node". By gateway node, do you mean the ingestion job coordinator node? `makePlan()` here is run within `Resume`.
- and the benefit of this patch is to ensure the frontier processor can act as normal, if the job lease transfers to another node, i.e., a different node begins to coordinate the job, correct? Could that ever happen without going through Resume?

Reply:
>  By gateway node, do you mean the ingestion job coordinator node?

Yes, but gateway node I mean whatever node is starting the DistSQL flow, which in this context will currently be the job coordinator.

> and the benefit of this patch is to ensure the frontier processor can act as normal, if the job lease transfers to another node, i.e., a different node begins to coordinate the job, correct? Could that ever happen without going through Resume?

The point of this is to prevent this node from updating the job progress when it has actually lost the lease on the job.  Another node would indeed have to go through Resume to start this DistSQL flow, but that can happen before the original job coordinator knows it has lost the lease.

Reply:
> Another node would indeed have to go through Resume to start this DistSQL flow, but that can happen before the original job coordinator knows it has lost the lease.

Ah, i see. I guess this is a problem with all jobs, but with stream ingestion, the problem is potentially more likely to occur because a distsql processor is updating the jobs table instead of some vanilla goroutine spun off by Resume (like in Restore).